webui/src/pages/extension-detail/extension-detail-overview.tsx

```diff
@@ -108,7 +108,7 @@ class ExtensionDetailOverviewComponent extends React.Component<ExtensionDetailOv
                 const readme = await this.props.service.getExtensionReadme(this.props.extension);
                 this.setState({ readme, loading: false });
             } catch (err) {
-                handleError(err);
+                this.props.setError(handleError(err));
```

Comment:
You could take care of converting the error to a string in the `setError` function so you don't need to call `handleError` in every subcomponent.